operator/pkg/status/kyma_handler.go

```diff
@@ -109,70 +97,21 @@ func (k *Kyma) GetTemplateInfoForModule(
 	return nil, ErrTemplateNotFound
 }
 
-func (k *Kyma) GetReadyConditionForComponent(kymaObj *operatorv1alpha1.Kyma,
-	componentName string,
-) (*operatorv1alpha1.KymaCondition, bool) {
-	status := &kymaObj.Status
-	for i := range status.Conditions {
-		existingCondition := &status.Conditions[i]
-		if existingCondition.Type == operatorv1alpha1.ConditionTypeReady && existingCondition.Reason == componentName {
-			return existingCondition, true
-		}
+func (k *Kyma) getModuleState(module *unstructured.Unstructured) operatorv1alpha1.State {
+	state, found, err := unstructured.NestedString(module.Object, watch.Status, watch.State)
+	if !found {
+		return operatorv1alpha1.StateProcessing
 	}
-
-	return &operatorv1alpha1.KymaCondition{
-		Type:   operatorv1alpha1.ConditionTypeReady,
-		Reason: componentName,
-	}, false
-}
-
-func (k *Kyma) UpdateConditionFromComponentState(name string, module *parsed.Module,
-	kyma *operatorv1alpha1.Kyma,
-) (bool, error) {
-	updateRequired := false
-	component := module.Unstructured
-	moduleName := component.GetLabels()[operatorv1alpha1.ModuleName]
-
-	status := component.Object[watch.Status]
-	if status != nil {
-		condition, exists := k.GetReadyConditionForComponent(kyma, moduleName)
-		if !exists {
-			// In rare cases, the condition for a module does not yet exist in an update
-			// this can happen if the creation failed due to a resource version conflict
-			// if this happens we need to make sure to create it dynamically
-			k.SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-				operatorv1alpha1.ConditionStatusFalse, "component tracked!")
-			updateRequired = true
-			return updateRequired, nil
-		}
-
-		switch status.(map[string]interface{})[watch.State].(string) {
-		case string(operatorv1alpha1.StateReady):
-			if condition.Status != operatorv1alpha1.ConditionStatusTrue {
-				k.SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-					operatorv1alpha1.ConditionStatusTrue, "component ready!")
-
-				updateRequired = true
-			}
-
-		case "":
-			if condition.Status != operatorv1alpha1.ConditionStatusUnknown {
-				k.SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-					operatorv1alpha1.ConditionStatusUnknown, "component status not known!")
-
-				updateRequired = true
-			}
-
-		default:
-			// This will only trigger an update on the conditions in case the component was not set to not false before
-			if condition.Status != operatorv1alpha1.ConditionStatusFalse {
-				k.SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-					operatorv1alpha1.ConditionStatusFalse, "component not ready!")
-
-				updateRequired = true
-			}
-		}
+	if err == nil && isValidState(state) {
+		return operatorv1alpha1.State(state)
 	}
+	return operatorv1alpha1.StateError
+}
 
-	return updateRequired, nil
+func isValidState(state string) bool {
```

Comment:
I think this should move into the machinery pkg later on. For now, how about we move it to the API package?

Reply:
Just want to clarify `machinery` pkg you mentioned here first. 
I see two options:
1.  refactoring the package structure similar to `k8s.io/apimachinery` and `k8s.io/api`, then `/api` only have resource type definition, all other existing code in the current `/api` can be reorganized under `/apimachinery/pkg/` 
2. only have `/apimachinery`, the type definition go to `/apimachinery/pkg/api`
Or do you have different view in your mind?

Reply:
And yes, this code should definitely not be here.

Reply:
I think I just leave this here, ideally `SyncModuleInfo` should also go to apimachinery.